pkgs/by-name/cr/cryptomator-cli/package.nix

```
@@ -0,0 +1,114 @@
+{
+  autoPatchelfHook,
+  fetchFromGitHub,
+  fuse3,
+  glib,
+  zulu25,
+  lib,
+  makeShellWrapper,
+  maven,
+  nix-update-script,
+}:
+
+let
+  jdk = zulu25;
+in
+maven.buildMavenPackage rec {
+  pname = "cryptomator-cli";
+  version = "0.6.2";
+
+  src = fetchFromGitHub {
+    owner = "cryptomator";
+    repo = "cli";
+    rev = version;
+    hash = "sha256-rwARleKktGXmumIBmrPrfls4EywBqGBxOaB8/ka5ds0=";
+  };
+
+  mvnJdk = jdk;
+  mvnParameters = "-Dmaven.test.skip=true";
+  mvnHash = "sha256-54DT4C+WzyUBPxayA9YnB9I/Igd19iZygByUh5of51I=";
+
+  preBuild = ''
+    export APP_VERSION=${version}
+    export SEMVER_STR=${version}
+  '';
+
+  # Based on the build_linux.sh script and jpackage configuration
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin/ $out/share/cryptomator-cli/libs/ $out/share/cryptomator-cli/mods/
+    mkdir -p $out/share/bash-completion/completions/
+
+    # Copy dependencies
+    cp target/libs/* $out/share/cryptomator-cli/libs/
+    cp target/mods/* target/cryptomator-cli-*.jar $out/share/cryptomator-cli/mods/
+
+    # Copy bash completion script if it exists
+    if [ -f target/cryptomator-cli_completion.sh ]; then
+      cp target/cryptomator-cli_completion.sh $out/share/bash-completion/completions/cryptomator-cli
+    fi
+
+    # Determine native access package based on architecture
+    NATIVE_ACCESS_PACKAGE="no.native.access.available"
+    if [ "$(uname -m)" = "x86_64" ]; then
+      NATIVE_ACCESS_PACKAGE="org.cryptomator.jfuse.linux.amd64"
+    elif [ "$(uname -m)" = "aarch64" ]; then
+      NATIVE_ACCESS_PACKAGE="org.cryptomator.jfuse.linux.aarch64"
+    fi
```

This is an interesting solution, most of the time architecture checks are done inside Nix rather than shell commands.
What about this?

```suggestion
    NATIVE_ACCESS_PACKAGE=${
      if stdenv.hostPlatform.isx86_64 then
        "org.cryptomator.jfuse.linux.amd64"
      else if stdenv.hostPlatform.isAarch64 then
        "org.cryptomator.jfuse.linux.aarch64"
      else
        "no.native.access.available"
    }
```